compiler/ccgreset.nim

```diff
@@ -87,6 +87,12 @@ proc specializeResetT(p: BProc, accessor: Rope, typ: PType) =
     lineCg(p, cpsStmts, "$1 = 0;$n", [accessor])
   of tyCstring, tyPointer, tyPtr, tyVar, tyLent:
     lineCg(p, cpsStmts, "$1 = NIM_NIL;$n", [accessor])
+  of tySet:
+    if mapSetType(p.config, typ) == ctArray:
+      lineCg(p, cpsStmts, "#nimZeroMem($1, sizeof($2));$n",
+          [accessor, getTypeDesc(p.module, typ)])
+    else:
+      lineCg(p, cpsStmts, "$1 = 0;$n", [accessor])
```

Comment:
the discard below is the problem really - ie it would be better if it listed the uninteresting types explicitly..

Reply:
I just followed the pre-existing patterns though

```nim
  of tySet:
    if mapSetType(p.config, ty) == ctArray:
      linefmt(p, cpsStmts, "#nimCopyMem((void*)$1, (NIM_CONST void*)$2, $3);$n",
              [rdLoc(dest), rdLoc(src), getSize(p.config, dest.t)])
    else:
      linefmt(p, cpsStmts, "$1 = $2;$n", [rdLoc(dest), rdLoc(src)])
```

Reply:
I just followed the pre-existing pattern though

```nim
  of tySet:
    if mapSetType(p.config, ty) == ctArray:
      linefmt(p, cpsStmts, "#nimCopyMem((void*)$1, (NIM_CONST void*)$2, $3);$n",
              [rdLoc(dest), rdLoc(src), getSize(p.config, dest.t)])
    else:
      linefmt(p, cpsStmts, "$1 = $2;$n", [rdLoc(dest), rdLoc(src)])
```

Reply:
Yeah, I prefer this better code.

Reply:
not sure I follow, ie the suggestion is to change `else: discard` to `of A, B, ...: discard` such that it's obvious which types are skipped by the resetter

Reply:
Sorry, I misunstood what you said. I thought you mean I didn't explicitly list the return type kind of mapSetType.


Yeah I agree.

Reply:
Thanks for your advice!